vm/src/stdlib/time.rs

```diff
@@ -173,6 +197,179 @@ fn time_strptime(
     Ok(PyStructTime::new(vm, instant, -1))
 }
 
+fn get_thread_time(vm: &VirtualMachine) -> PyResult<Duration> {
+    cfg_if::cfg_if! {
+        if #[cfg(windows)] {
+            let mut _creation_time = FILETIME::default();
+            let mut _exit_time = FILETIME::default();
+            let mut kernel_time = FILETIME::default();
+            let mut user_time = FILETIME::default();
+
+            let (ktime, utime) = unsafe {
+                let thread = GetCurrentThread();
+                if GetThreadTimes(thread, &mut _creation_time, &mut _exit_time, &mut kernel_time, &mut user_time) == 0 {
+                    return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+                }
+                (time_from_filtime(kernel_time), time_from_filtime(user_time))
+            };
+            Ok(Duration::from_nanos((ktime + utime) * 100))
+        } else if #[cfg(any(
+            target_os = "android",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "linux",
+            target_os = "openbsd",
+        ))] {
+            let mut time = libc::timespec {
+                tv_sec: 0,
+                tv_nsec: 0,
+            };
+            if unsafe { libc::clock_gettime(libc::CLOCK_THREAD_CPUTIME_ID, &mut time) } == -1 {
+                return Err(vm.new_os_error("Failed to get clock time".to_owned()));
+            }
+
+            Ok(Duration::new(time.tv_sec as u64, time.tv_nsec as u32))
+        } else if #[cfg(solaris)] {
+            Ok(Duration::from_nanos(unsafe { libc::gethrvtime() }))
+        } else {
+            Err(vm.new_not_implemented_error("thread time unsupported in this system".to_owned()))
+        }
+    }
+}
+
+fn time_thread_time(vm: &VirtualMachine) -> PyResult<f64> {
+    Ok(get_thread_time(vm)?.as_secs_f64())
+}
+
+fn time_thread_time_ns(vm: &VirtualMachine) -> PyResult<u64> {
+    Ok(get_thread_time(vm)?.as_nanos() as u64)
+}
+
+#[cfg(windows)]
+unsafe fn time_from_filtime(time: FILETIME) -> u64 {
```

Comment:
I missed this part. `unsafe` keyword has different meaning for `unsafe fn` and `unsafe {}`. For `unsafe fn`, it means calling the fn is not safe. For `unsafe {}`, it mean the block is using unsafe operations. So in this case, each function calls are unsafe but the function itself is safe.